hail/python/hailtop/utils/utils.py

```diff
@@ -882,12 +882,16 @@ async def run_if_changed_idempotent(changed, f, *args, **kwargs):
 
 
 async def periodically_call(period, f, *args, **kwargs):
-    async def loop():
-        log.info(f'starting loop for {f.__name__}')
-        while True:
+    log.info(f'starting loop for {f.__name__}')
+    while True:
+        try:
             await f(*args, **kwargs)
+        except asyncio.CancelledError:
+            raise
+        except Exception:
+            log.exception(f'in {f.__name__}')
+        finally:
             await asyncio.sleep(period)
```

Comment:
I think this will sleep even if a `CancelledError` was raised, which I don't think we want. I also think this suffers from a similar problem of putting logic in multiple different places. I think we should push any specific error handling closer to where that error might occur, so whatever function we pass to `periodically_call`. I don't think `periodically_call` has to change here if we can properly suppress errors inside of our resource-usage collecting function.

Reply:
I abandoned fixing periodically call at all and pushed the retry logic to the `measure` function.

Reply:
Just to be clear. The semantics I want are to never execute that function more than once every 5 seconds no matter what. This way we guarantee that there's no possible way to fill the disk with writes more than every 5 seconds.